### ISSUE TITLE
Setup staging step

### ISSUE THREAD

**Issue by zadigus** [shopozor/backend#70]
Create staging jenkins job that

* builds production docker image
* deploys the production image to the staging environment
* performs tests on environment setup on the staging backend environment (e.g. check that debug is disabled, allowed hosts is defined correctly, etc.)
* start test client environment that performs basic API tests (in order to verify that the server is really up and running and graphql is working)

If the tests on the staging env pass, then

* tag the master branch
* deploy tag to production environment
* send me an e-mail that the tag is successfully tested

After that, I may want to create a release on github manually.

**Comment by shikamu** [comment]
security checks:

1. run our site through https://securityheaders.com, the only acceptable grade is A+. This can be automated by using curl, e.g. if our domain is softozor.ch, we could run **curl -I -X HEAD https://securityheaders.com/?q=https%3A%2F%2Fsoftozor.ch&followRedirects=on**. This request returns two specific headers, one called **x-grade** and the other called **x-score**. x-score is a base64 string that contains some information, at the moment not a lot of them, and x-grade is a string that contains the grade which is, at the time of writing, any of ("A+", "A", "B", "C", "D", "E", "F"). We want A+

2. run our site through ssllabs.com, e.g. https://www.ssllabs.com/ssltest/analyze.html?d=softozor.ch
There is also apparently an api project for this, see https://www.ssllabs.com/projects/ssllabs-apis/  and the github project is https://github.com/ssllabs/ssllabs-scan/ it's written in the Go language.
I've not used this tool so we would need to try it out. I'm not sure what kind of output the tool yields, but from a manual run on the website, the only acceptable grade is again A+.

**Comment by zadigus** [comment]
@shikamu would it make sense to regularly (e.g. once a week) run the security checks you're mentioning on the production environment too? because I can imagine that those checks get updated with the latest knowledge on the topic and that a web site with "A+" grade today might have "B" grade tomorrow. What do you think? What security tests would then make sense on the production environment on a regular basis?

**Comment by shikamu** [comment]
you are absolutely right. Ideally we should run some sort of cron task to run the above 2 checks and get notified (by email for example) of the results. It would then obviously be up to us, in particular me, to act upon the notification.

**Comment by zadigus** [comment]
#76
